app/controllers/provider_interface/provider_users_controller.rb

```diff
@@ -33,15 +39,42 @@ def create
       redirect_to provider_interface_provider_users_path
     end
 
+    def edit_providers
+      provider_user = ProviderUser.find(params[:provider_user_id])
+      @form = ProviderUserForm.from_provider_user(provider_user)
+      @form.current_provider_user = current_provider_user
+    end
+
+    def update_providers
+      provider_user = ProviderUser.find(params[:provider_user_id])
+      provider_user.assign_attributes(provider_user_params.except(:permissions))
+      @form = ProviderUserForm.from_provider_user(provider_user)
+      service = SaveProviderUser.new(provider_user: provider_user, permissions: permissions_params)
+
+      render :edit_providers and return unless service.call!
```

Comment:
This will still raise a 500 `ActionController::ParameterMissing` if no checkbox is checked on the edit providers page.

Reply:
Thanks @tijmenb this is cause by the checkboxes no being included in the form post requests if they are blank. Played around with this a bit this morning and came up with the following fix:

```ruby
    def update_providers
      provider_user = ProviderUser.find(params[:provider_user_id])

      redirect_to_edit_permissions_with_warning and return if provider_user_params.nil?

      provider_user.assign_attributes(provider_user_params.except(:permissions))
      @form = ProviderUserForm.from_provider_user(provider_user)
      service = SaveProviderUser.new(provider_user: provider_user, permissions: permissions_params)

      render :edit_providers and return unless service.call!

      flash[:success] = 'Providers updated'
      redirect_to provider_interface_provider_user_path(provider_user)
    end

  private

    def provider_user_params
      if params.has_key? (:provider_interface_provider_user_form)
        params.require(:provider_interface_provider_user_form)
              .permit(:email_address, :first_name, :last_name, provider_ids: [], permissions: {})
      end
    end

    def redirect_to_edit_permissions_with_warning
      flash[:warning] = 'A user needs to be associated with at least one provider'
      redirect_to provider_interface_provider_user_edit_providers_path(params[:provider_user_id])
    end
```
My understanding is that we would want users to be associated with at least one provider hence the use of the warning.

I think this is a bit messy though - the other option I was thinking about is to have hidden fields that submit with the checkboxes. This means that the form params can still be retrieved from the checkbox form and then I could add validations to the model rather than special control flow in the controller.

What are your thoughts?

Working on a spec for this in any case as the result should be the same I think.

Reply:
Ignore this I was acting on old information 😢